lib/wakes.rb

```diff
@@ -39,8 +39,12 @@ def self.create!(*args)
     build(*args).tap(&:save!)
   end
 
-  def self.redirect(source, target, label = nil)
-    RedirectMapper.new(source, target, label)
+  def self.redirect(source_path, target_path, label = nil)
+    RedirectMapper.new(source_path, target_path, label)
+  end
+
+  def self.redirect_with_hosts(source_host: nil, source_path:, target_host: nil, target_path:, label: nil)
```

Comment:
@chrismaddocks I'm not a fan of this API.

Why not just use `self.redirect(source_path_or_url, target_path_or_url, label = nil)`?

Then add in logic in `RedirectMapper` to determine if a host is present or not.

A few reasons:

1.) It's a nicer API.
2.) It means that we don't have to maintain two different APIs.
3.) It means that devs don't have to remember two different APIs.